Docs: Add Whats new in 7.5 (#31659)
grafana/grafana: docs/sources/whatsnew/whats-new-in-v7-5.md

+++
title = "What's new in Grafana v7.5"
description = "Feature and improvement highlights for Grafana v7.5"
keywords = ["grafana", "new", "documentation", "7.5", "release notes"]
weight = -32
[_build]
list = false
+++

# What’s new in Grafana v7.5

> **Note:** This is content for a beta version of Grafana. It will be updated several times before the stable release.

> This topic includes the release notes for Grafana v7.5. For all details, read the full [CHANGELOG.md](https://github.com/grafana/grafana/blob/master/CHANGELOG.md).

Reviewer: ```suggestion
This topic discusses the new features and enhancements in Grafana v7.5 release. For all details, read the full [CHANGELOG.md](https://github.com/grafana/grafana/blob/master/CHANGELOG.md).
```

With alerting support for Loki, you can now create alerts on Loki metrics queries. Alert rules can be set up easily in the graph panel and you can learn more about the process of creating alerts in the Alert documentation.

![Loki alerting](/img/docs/alerting/alerting-for-loki-7-5.png)

### Loki label browser

A new Loki logs browser lets you construct the queries step by step: you choose labels that you like to consider, such as "job", then you select the values that you like, such as "my-app1". Note that you can select values from more than one label, and that they get facetted. This means only possible label combinations are selectable. When you're done, you can run the query either as a logs or a metrics query (a metrics query returns the log volume in a chart).

### Changed default HTTP method for new Prometheus data sources

> For new Prometheus data sources, we have changed the default HTTP method to POST. POST allows for much larger query bodies than using the GET method. This is necessary when sending queries from graphs with a lot of targets, such as many hosts in a dashboard variable. POST also makes the Query Inspector data easier to read since the query is in plain text where the GET query is URL encoded.

Reviewer: ```suggestion
For new Prometheus data sources, we have changed the default HTTP method to POST. POST allows for much larger query bodies than using the GET method. This is necessary when sending queries from graphs with a lot of targets, for example, many hosts in a dashboard variable. The POST method also makes the Query Inspector data easier to read since the query is in plain text whereas the GET query is URL encoded.
```

> Note: This is not going to affect provisioned data sources or already created data sources.

Reviewer: ```suggestion
>**Note:** This is not going to affect provisioned data sources or already created data sources.
```

### Word highlighting for Elasticsearch

When searching for text in Elasticsearch logs, matching words in the log line returned by the query are now highlighted.

![Elastic logs highlighting](/img/docs/elastic/elastic-word-highlighting-7-5.png)

### Better format definition for trace data

In Grafana 7.5, we changed how data for the trace view is sent from the data source. The required data frame has a clear format, which is more aligned with how data is generally represented in Grafana. This makes it easier for third-party developers to implement data sources leveraging the trace view.

### Paste in SSL certs for Postgres data source

Previously, when users wanted to configure the Postgres data source to connect with SSL certification, they needed to put the certification on the server, and configure the data source with file path.

> With the current feature, instead of file path, users can just paste the SSL certification content in the UI. This allows users to configure the certification without access to the server.

Reviewer: ```suggestion
Instead of the file path, users can now paste the SSL certification content in the UI. This allows them to configure the certification even when they do not have access to the server.
```

> **Note:** It remains as limitation for the hosted Grafana, because the user doesn't have access to the server configuration.

### Deprecation notice for some Azure Monitor queries 

In the upcoming Grafana 8.0 release, Application Insights and Insights Analytics query types within the Azure Monitor data source will be deprecated and be made read-only in favor of querying Application Insights from Metrics and Logs. 

> Grafana 7.5 includes a deprecation notice for Application Insights and Insights Analytics queries, and some documentation to help users prepare for the upcoming changes.

Reviewer: ```suggestion
Grafana 7.5 includes a deprecation notice for these queries, and some documentation to help users prepare for the upcoming changes.
```
